modules/sdk-coin-ada/test/unit/transactionBuilder.ts

```diff
@@ -47,25 +51,181 @@ describe('ADA Transaction Builder', async () => {
+      'https://testnet.koios.rest/api/v0/address_info?_address=' + txBuilder.getChangeAddress
+    );
+    const senderBalance = utxoData.data[0].balance;
+    txBuilder.senderBalance(senderBalance);
+    txBuilder.ttl(800000000);
+    txBuilder.sign({ key: testData.privateKeys.prvKey4 });
+  });
+
   it('build a send from rawTx', async () => {
     const txBuilder = factory.from(testData.rawTx.signedTx);
     const builtTx = await txBuilder.build();
     should.equal(builtTx.type, TransactionType.Send);
-    should.equal(builtTx.id, testData.rawTx.txHash);
-    builtTx.inputs.length.should.equal(0);
-    builtTx.outputs.length.should.equal(2);
-    builtTx.outputs[0].should.deepEqual(testData.rawTx.outputAddress1);
-    builtTx.outputs[1].should.deepEqual(testData.rawTx.outputAddress2);
+    should.equal(builtTx.id, testData.rawTx.txHash2);
+  });
+
+  it('should successfully submit a transaction', async () => {
+    const keyPair = new KeyPair({ prv: testData.privateKeys.prvKey4 });
+    const senderAddress = keyPair.getAddress(AddressFormat.testnet);
+    const axiosConfig = {
+      headers: {
+        'Content-Type': 'application/cbor',
+      },
+      timeout: 10000,
+    };
+
+    const txBuilder = factory.getTransferBuilder();
+    txBuilder.changeAddress(senderAddress);
+    const utxoData = await axios.get('https://testnet.koios.rest/api/v0/address_info?_address=' + senderAddress);
+    // need to add in change address
+    const senderBalance = utxoData.data[0].balance;
+    txBuilder.senderBalance(senderBalance);
+    const utxoSet = utxoData.data[0].utxo_set;
+    for (const utxo of utxoSet) {
+      txBuilder.input({ transaction_id: utxo.tx_hash, transaction_index: utxo.tx_index });
+    }
+
+    txBuilder.output({
+      address:
+        'addr_test1qz2fxv2umyhttkxyxp8x0dlpdt3k6cwng5pxj3jhsydzer3jcu5d8ps7zex2k2xt3uqxgjqnnj83ws8lhrn648jjxtwq2ytjqp',
+      amount: '5000000',
+    });
+
+    txBuilder.ttl(800000000);
+    txBuilder.sign({ key: keyPair.getKeys().prv });
+
+    const tx = await txBuilder.build();
+    const serializedTx = tx.toBroadcastFormat();
+    const bytes = Uint8Array.from(Buffer.from(serializedTx, 'hex'));
+
+    try {
+      const res = await axios.post('https://testnet.koios.rest/api/v0/submittx', bytes, axiosConfig);
+      console.log(res.data);
+    } catch (err) {
+      console.log(err);
+    }
   });
 
   it('build a send from unsigned rawTx', async () => {
     const txBuilder = factory.from(testData.rawTx.unsignedTx);
     const builtTx = await txBuilder.build();
     should.equal(builtTx.type, TransactionType.Send);
     should.equal(builtTx.id, testData.rawTx.txHash);
-    builtTx.inputs.length.should.equal(0);
     builtTx.outputs.length.should.equal(2);
     builtTx.outputs[0].should.deepEqual(testData.rawTx.outputAddress1);
     builtTx.outputs[1].should.deepEqual(testData.rawTx.outputAddress2);
   });
+
+  it('should use the addSignature interface', async () => {
```

Comment:
the test that use koios has to be removed and move to sandbox because the test are going to run by CI and should not access outside network

Reply:
got it, will remove